crates/re_chunk_store/src/writes.rs

```diff
@@ -0,0 +1,164 @@
+    ///
+    /// Iff the store was modified, all registered subscribers will be notified and the
+    /// resulting [`ChunkStoreEvent`] will be returned, or `None` otherwise.
+    ///
+    /// * Trying to insert an unsorted chunk ([`Chunk::is_sorted`]) will fail with an error.
+    /// * Inserting a duplicated [`ChunkId`] will result in a no-op.
+    /// * Inserting an empty [`Chunk`] will result in a no-op.
+    pub fn insert_chunk(
+        &mut self,
+        chunk: &Arc<Chunk>,
+    ) -> ChunkStoreResult<Option<ChunkStoreEvent>> {
+        if self.chunks_per_chunk_id.contains_key(&chunk.id()) {
+            // We assume that chunk IDs are unique, and that reinserting a chunk has no effect.
+            re_log::warn_once!(
+                "Chunk #{} was inserted more than once (this has no effect)",
+                chunk.id()
+            );
+            return Ok(None);
+        }
+
+        if !chunk.is_sorted() {
+            return Err(ChunkStoreError::UnsortedChunk);
+        }
+
+        let Some(row_id_range) = chunk.row_id_range() else {
+            return Ok(None);
+        };
+
+        re_tracing::profile_function!(format!("{}", row_id_range.0));
+
+        self.insert_id += 1;
+
+        let row_id_min = row_id_range.0;
+        let row_id_max = row_id_range.1;
+
+        self.chunks_per_chunk_id.insert(chunk.id(), chunk.clone());
+        self.chunk_ids_per_min_row_id
+            .entry(row_id_min)
+            .or_default()
+            .push(chunk.id());
+
+        if chunk.is_static() {
+            // Static data: make sure to keep the most recent chunk available for each component column.
+
+            for (&component_name, list_array) in chunk.components() {
+                let is_empty = list_array
+                    .validity()
+                    .map_or(false, |validity| validity.is_empty());
+                if is_empty {
+                    continue;
+                }
+
+                self.static_chunk_ids_per_entity
+                    .entry(chunk.entity_path().clone())
+                    .or_default()
+                    .entry(component_name)
+                    .and_modify(|cur_chunk_id| {
+                        // NOTE: When attempting to overwrite static data, the chunk with the most
+                        // recent data within -- according to RowId -- wins.
+
+                        let cur_row_id_max = self.chunks_per_chunk_id.get(cur_chunk_id).map_or(
+                            RowId::ZERO,
+                            |chunk| {
+                                chunk
+                                    .row_id_range()
+                                    .map_or(RowId::ZERO, |(_, row_id_max)| row_id_max)
+                            },
+                        );
+                        if row_id_max > cur_row_id_max {
+                            *cur_chunk_id = chunk.id();
+                        }
+                    })
+                    .or_insert_with(|| chunk.id());
```

Comment:
:warning:

That's the one place where we are purposefully (and rightfully) creating a (potentially) dangling chunk by overwriting the old ID.

That's fine in and of itself, but we need to tweak the GC code so that it handles dangling static chunks properly.